tests/model_inheritance/models.py

```diff
@@ -186,3 +186,23 @@ class Child(Parent):
 
 class GrandChild(Child):
     pass
+
+
+class CommonAncestor(models.Model):
+    id = models.IntegerField(primary_key=True, default=1)
+
+
+class FirstParent(CommonAncestor):
+    first_ancestor = models.OneToOneField(
+        CommonAncestor, models.CASCADE, primary_key=True, parent_link=True
+    )
+
+
+class SecondParent(CommonAncestor):
+    second_ancestor = models.OneToOneField(
+        CommonAncestor, models.CASCADE, primary_key=True, parent_link=True
+    )
+
+
+class CommonChild(FirstParent, SecondParent):
+    pass
```

Comment:
I tried to test this with more complicated inheritance, e.g.
```mermaid
graph TD;
    Place --> Restaurant
    Place --> ItalianRestaurantManyParents
    Rating --> Restaurant
    Restaurant --> ItalianRestaurant
    ItalianRestaurant --> ItalianRestaurantManyParents
```
Unfortunately, it crashes:
```
  File "/django/tests/model_inheritance/tests.py", line 164, in test_create_diamon_mti_different_levels
    ItalianRestaurantManyParents.objects.create(
  File "/django/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/django/django/db/models/query.py", line 650, in create
    obj.save(force_insert=True, using=self.db)
  File "/django/django/db/models/base.py", line 814, in save
    self.save_base(
  File "/django/django/db/models/base.py", line 876, in save_base
    parent_inserted = self._save_parents(cls, using, update_fields)
  File "/django/django/db/models/base.py", line 937, in _save_parents
    setattr(self, field.attname, self._get_pk_val(parent._meta))
AttributeError: 'OneToOneField' object has no attribute 'attname'
```
This is a separate issue, see ticket-34634.

Reply:
I'd prefer to fix 34634 first as this patch will remove unnecessary update in such cases as well.